tests/test_utils.py

```diff
@@ -290,13 +299,381 @@ def test_random_interferometer(self):
+
+    def test_extract_unitary_1_mode_tf(self):
+        """Test that unitary extraction works for 1 mode using the tf backend"""
+        self.logTestName()
+        eng_extract, q_extract = sf.Engine(num_subsystems=1)
+
+        S = sf.ops.Sgate(0.4, -1.2)
+        D = sf.ops.Dgate(2, 0.9)
+        K = sf.ops.Kgate(-1.5)
+
+        initial_state = np.random.rand(self.D) + 1j*np.random.rand(self.D) # not a state but it doesn't matter
+
+        with self.eng_sf:
+            sf.ops.Ket(initial_state) | self.q_sf
+            S | self.q_sf
+            D | self.q_sf
+            K | self.q_sf
+
+        with eng_extract:
+            S | q_extract
+            D | q_extract
+            K | q_extract
+
+        matrix_extract_tf_backend = extract_unitary(eng_extract, cutoff_dim=self.D, backend='tf')
+
+        with tf.Session() as sess:
+            sess.run(tf.global_variables_initializer())
+            in_state = tf.constant(initial_state.reshape([-1]), dtype=tf.complex64)
+            final_state_tf_backend = sess.run(tf.einsum('ab,b', matrix_extract_tf_backend, in_state))
+
+        final_state_sf = self.eng_sf.run().ket()
+
+        self.assertAllAlmostEqual(final_state_tf_backend, final_state_sf, delta=self.tol)
+
+    def test_extract_channel_1_mode(self):
+        """Test that channel extraction works for 1 mode"""
+        self.logTestName()
+        eng_extract, q_extract = sf.Engine(num_subsystems=1)
+
+        S = sf.ops.Sgate(1.1, -1.4)
+        L = sf.ops.LossChannel(0.45)
+
+        initial_state = np.random.rand(self.D, self.D)
+
+        with self.eng_sf:
+            sf.ops.DensityMatrix(initial_state) | self.q_sf
+            S | self.q_sf
+            L | self.q_sf
+
+        with eng_extract:
+            S | q_extract
+            L | q_extract
+
+        choi = extract_channel(eng_extract, cutoff_dim=self.D, vectorize_modes=True, representation='choi')
+        liouville = extract_channel(eng_extract, cutoff_dim=self.D, vectorize_modes=True, representation='liouville')
+        kraus = extract_channel(eng_extract, cutoff_dim=self.D, vectorize_modes=True, representation='kraus')
+
+        final_rho_sf = self.eng_sf.run().dm()
+
+        final_rho_choi = np.einsum('abcd,ab -> cd', choi, initial_state)
+        final_rho_liouville = np.einsum('abcd,db -> ca', liouville, initial_state)
+        final_rho_kraus = np.einsum('abc,cd,aed -> be', kraus, initial_state, np.conj(kraus))
+
+        self.assertAllAlmostEqual(final_rho_choi, final_rho_sf, delta=self.tol)
+        self.assertAllAlmostEqual(final_rho_liouville, final_rho_sf, delta=self.tol)
+        self.assertAllAlmostEqual(final_rho_kraus, final_rho_sf, delta=self.tol)
+
+    def test_vectorize_unvectorize(self):
+        """Test vectorize utility function"""
+        self.logTestName()
+
+        cutoff = 4
+
+        dm = np.random.rand(*[cutoff]*8) + 1j*np.random.rand(*[cutoff]*8) # 4^8 -> (4^2)^4 -> 4^8
```

Comment:
make (arbitrary) choices here instead of random

Reply:
I set the seed after importing numpy